examples/api/lib/material/dropdown_menu/dropdown_menu.0.dart

```diff
@@ -34,41 +56,55 @@ class DropdownMenuExample extends StatelessWidget {
       ),
       home: Scaffold(
         body: SafeArea(
-          child: Padding(
-            padding: const EdgeInsets.only(top: 20),
-            child: Row(
-              mainAxisAlignment: MainAxisAlignment.center,
-              children: <Widget>[
-                DropdownMenu(
-                  label: const Text('Label'),
-                  dropdownMenuEntries: dropdownMenuEntries,
+          child: Column(
+            children: <Widget>[
+              Padding(
+                padding: const EdgeInsets.symmetric(vertical: 20),
+                child: Row(
+                  mainAxisAlignment: MainAxisAlignment.center,
+                  children: <Widget>[
+                    DropdownMenu(
+                      controller: colorController,
+                      label: const Text('Color'),
+                      dropdownMenuEntries: colorEntries,
+                      onChanged: (String text) {
+                        setState(() {
+                          hasColor = colors.containsKey(text);
+                        });
+                      },
+                    ),
+                    const SizedBox(width: 20),
+                    DropdownMenu(
+                      controller: iconController,
+                      enableFilter: true,
+                      leadingIcon: const Icon(Icons.search),
+                      label: const Text('Icon'),
+                      dropdownMenuEntries: iconEntries,
+                      inputDecorationTheme: const InputDecorationTheme(filled: true),
+                      onChanged: (String text) {
```

Comment:
Although this approach could be made to work - and it avoids giving DropdownMenu a type parameter - it's not going to be convenient for apps that have localized the menu labels. We could take the same approach as DropdownButton: 
```dart
DropdownMenu<T> has-a list of DropdownMenuEntry<T>
DropdownMenuEntry<T> has a value property of type T
DropdownMenu's onChanged callback is ValueChanged<T?>? (null means the menu was dismissed with no selection)
```

Reply:
Note also: perhaps this callback should be called "onSelected" to leave room for an onChanged callback that runs each time the textfield changes. Some apps might want that, e.g. to deal with typos or whatever. Not saying that it will necessarily be a good idea, but [DWIM](https://github.com/flutter/flutter/issues/115501#issuecomment-1332448442)

Reply:
I agree with both of these comments. A typed parameter would also make it more consistent with SegmentedButton as well.

+1 for calling it 'onSelected' and having it called only when the selection changes, and not every key stroke. If they need that granularity, they can add a listener to the text controller.

Reply:
I see. Thanks so much for the suggestions. I added a generic type for both DropdownMenu and DropdownMenuEntry. Please let me know if there's any problems.